File: ios/scm/Models/Evidence.swift
//
//  Evidence.swift
//  scm
//
//  Created by Kemal Dwi Heldy Muhammad on 05/02/26.
//

struct Evidence: Codable, Identifiable {
    var id: String
    var cluster_id: String
    var processor_id: String
    var camera_id: String
    var frame_id: String
    var timestamp: Int64
    var person: [EvidencePerson]
}

struct EvidencePerson: Hashable, Codable {
    var id: String
    var bbox: [Float]
    var confidence: Float
    var part: [EvidencePersonPart]
    var equipment: [EvidencePersonEquipment]
    var violation: [EvidencePersonViolation]
}

struct EvidencePersonPart: Hashable, Codable {
    var label: EvidencePersonPartLabel
    var bbox: [Float]
    var confidence: Float
}
enum EvidencePersonPartLabel: String, Equatable, Codable {
    case head
    case hand
    case face
    case foot
    case ear
}

struct EvidencePersonEquipment: Hashable, Codable {
    var label: EvidencePersonEquipmentLabel
    var bbox: [Float]
    var confidence: Float
}
enum EvidencePersonEquipmentLabel: String, Equatable, Codable {
    case hardhat
    case gloves
    case shoes
    case safetyvest
    case safetysuit
    case facemask
    case faceguard
    case earmuffs
    case glasses
}

enum EvidencePersonViolation: String, Equatable, Codable {
    case missing_hardhat
    case missing_gloves
    case missing_shoes
    case missing_facemask
    case missing_earmuffs
    case missing_safetyvest
    case improperly_worn_hardhat
    case improperly_worn_gloves
    case improperly_worn_shoes
    case improperly_worn_facemask
    case improperly_worn_earmuffs
}

struct ViewEvidence: Hashable, Codable, Identifiable {
    var id: String
    var cluster: ClusterRef
    var processor: ProcessorRef
    var camera: CameraRef
    var timestamp: Int64
    var person: [EvidencePerson]
}

File: processor-web/app/types/evidence.ts
export type Evidence = {
  id: string;
  camera_id: string;
  frame_id: string;
  timestamp: number;
  person: EvidencePerson[];
};

export type EvidencePerson = {
  id: string;
  bbox: [number, number, number, number];
  confidence: number;
  part: EvidencePersonPart[];
  equipment: EvidencePersonEquipment[];
  violation: EvidencePersonViolation[];
};

export type EvidencePersonPart = {
  label: EvidencePersonPartLabel;
  bbox: [number, number, number, number];
  confidence: number;
};

export type EvidencePersonEquipment = {
  label: EvidencePersonEquipmentLabel;
  bbox: [number, number, number, number];
  confidence: number;
};

export type EvidencePersonPartLabel =
  | "head"
  | "hand"
  | "face"
  | "foot"
  | "ear";

export type EvidencePersonEquipmentLabel =
  | "hardhat"
  | "gloves"
  | "shoes"
  | "safetyvest"
  | "safetysuit"
  | "facemask"
  | "faceguard"
  | "earmuffs"
  | "glasses";

export type EvidencePersonViolation =
  | "missing_hardhat"
  | "missing_gloves"
  | "missing_shoes"
  | "missing_facemask"
  | "missing_earmuffs"
  | "missing_safetyvest"
  | "improperly_worn_hardhat"
  | "improperly_worn_gloves"
  | "improperly_worn_shoes"
  | "improperly_worn_facemask"
  | "improperly_worn_earmuffs";
